CefSharp.Core/Internals/CefCertificateCallbackWrapper.h

```diff
@@ -0,0 +1,65 @@
+// Copyright � 2010-2016 The CefSharp Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+#pragma once
+
+#include "Stdafx.h"
+
+#include "include\cef_request_handler.h"
+#include "CefWrapper.h"
+
+namespace CefSharp
+{
+	namespace Internals
+	{
+		private ref class CefCertificateCallbackWrapper : public ISelectClientCertificateCallback, public CefWrapper
+		{
+		private:
+			MCefRefPtr<CefSelectClientCertificateCallback> _callback;
+			const CefRequestHandler::X509CertificateList& _certificateList;
+
+		public:
+			CefCertificateCallbackWrapper(CefRefPtr<CefSelectClientCertificateCallback>& callback, const CefRequestHandler::X509CertificateList& certificates)
+				: _callback(callback), _certificateList(certificates)
+			{
+
+			}
+
+			!CefCertificateCallbackWrapper()
+			{
+				_callback = NULL;
+			}
+
+			~CefCertificateCallbackWrapper()
+			{
+				this->!CefCertificateCallbackWrapper();
+
+				_disposed = true;
+			}
+
+			virtual void Select(System::Security::Cryptography::X509Certificates::X509Certificate2^ cert)
+			{
+				ThrowIfDisposed();
+				if (!cert){
+					_callback->Select(NULL);
+				}
+
+				//Need cert_name from cert
+				auto cert_name = cert->SubjectName->Name;
```

Comment:
Read the binary data and use one of the `Encoding` classes to convert to a `string`. If you have to compare the serial, then you'd probably be better to convert the string you have and compare the two arrays.